fuel/converters/celeba.py

```diff
@@ -0,0 +1,212 @@
+    h5file = h5py.File(output_path, mode='w')
+    split_dict = {
+        'train': {
+            'features': (0, TRAIN_STOP),
+            'attributes': (0, TRAIN_STOP)},
+        'valid': {
+            'features': (TRAIN_STOP, VALID_STOP),
+            'attributes': (TRAIN_STOP, VALID_STOP)},
+        'test': {
+            'features': (VALID_STOP, NUM_EXAMPLES),
+            'attributes': (VALID_STOP, NUM_EXAMPLES)}}
+    h5file.attrs['split'] = H5PYDataset.create_split_array(split_dict)
+
+    attributes_dataset = h5file.create_dataset(
+        'attributes', (NUM_EXAMPLES, 40), dtype='uint8')
+    attributes_dataset.dims[0].label = 'batch'
+    attributes_dataset.dims[1].label = 'attribute'
+    attributes_dataset[...] = (
+        numpy.loadtxt(os.path.join(directory, ATTRIBUTES_FILE), dtype='int32',
+                      skiprows=2, usecols=tuple(range(1, 41))) +
+        1) / 2
+
+    features_dataset = h5file.create_dataset(
+        'features', (NUM_EXAMPLES, 3) + image_shape, dtype='uint8')
+    features_dataset.dims[0].label = 'batch'
+    features_dataset.dims[1].label = 'channel'
+    features_dataset.dims[2].label = 'height'
+    features_dataset.dims[3].label = 'width'
+
+    return h5file
+
+
+@check_exists(required_files=DATASET_FILES)
+def convert_celeba_aligned_cropped(directory, output_directory,
+                                   output_filename='celeba_aligned_cropped.hdf5'):
+    """Converts the aligned and cropped CelebA dataset to HDF5.
+
+    Converts the CelebA dataset to an HDF5 dataset compatible with
+    :class:`fuel.datasets.CelebA`. The converted dataset is saved as
+    'celeba_aligned_cropped.hdf5'.
+
+    It assumes the existence of the following files:
+
+    * `img_align_celeba.zip`
+    * `list_attr_celeba.txt`
+
+    Parameters
+    ----------
+    directory : str
+        Directory in which input files reside.
+    output_directory : str
+        Directory in which to save the converted dataset.
+    output_filename : str, optional
+        Name of the saved dataset. Defaults to 'celeba_aligned_cropped.hdf5'.
+
+    Returns
+    -------
+    output_paths : tuple of str
+        Single-element tuple containing the path to the converted dataset.
+
+    """
+    output_path = os.path.join(output_directory, output_filename)
+    h5file = _initialize_conversion(directory, output_path, (218, 178))
+
+    features_dataset = h5file['features']
+    image_file_path = os.path.join(directory, IMAGE_FILE)
+    with zipfile.ZipFile(image_file_path, 'r') as image_file:
+        with progress_bar('images', NUM_EXAMPLES) as bar:
+            for i in range(NUM_EXAMPLES):
+                image_name = 'img_align_celeba/{:06d}.jpg'.format(i + 1)
+                features_dataset[i] = numpy.asarray(
+                    Image.open(
+                        image_file.open(image_name, 'r'))).transpose(2, 0, 1)
+                bar.update(i + 1)
+
+    h5file.flush()
```

Comment:
I've seen this in a bunch of places. What's the point to `flush()` right before `close()`, doesn't the latter imply the former?